### DISCUSSION TITLE
When using MaterializeResult, how can I access a value or metadata in a downstream asset?

### DISCUSSION THREAD

**Discussion by seandavi** [dagster-io/dagster#18026] [Q&A] [answered]
When using [MaterializeResult](https://docs.dagster.io/_apidocs/assets#dagster.MaterializeResult), can I access the associated metadata (or a single value) in a downstream asset? I use the MaterializeResult approach since all of my assets are larger than memory and typically stored as a file in cloud storage or as a database table.

**Comment by seandavi** [comment, marked as answer]
This question looks like it has been asked and answered in a different context. https://github.com/dagster-io/dagster/discussions/14338

**Comment by oliverangelil** [comment]
You can do so using the `value` argument:

```
    return dg.MaterializeResult(
        value=df_agg,
        metadata={
            "Number of rows": dg.MetadataValue.int(len(df_agg))
        }
    )
```
